src/contracts.js

```diff
@@ -0,0 +1,68 @@
+import * as React from "react"
+import { Datagrid, DateField, EditButton, List, ReferenceField, TextField, DateInput, Edit, ReferenceInput, SelectInput, SimpleForm, NumberField, NumberInput,
+TextInput, Create } from 'react-admin';
+
+export const ContractList = () => (
+    <List>
+        <Datagrid rowClick="edit">
+            <ReferenceField source="companyId" reference="companies">
+                <TextField source="name" />
+            </ReferenceField>
+            <ReferenceField source="employeeId" reference="employees" label="Employee FN">
+                <TextField source="firstName" label="First Name"/>
+            </ReferenceField>
+            <ReferenceField source="employeeId" reference="employees" label="Employee LN" >
+                <TextField source="lastName" />
+            </ReferenceField>
+            <DateField source="startDate" />
+            <DateField source="endDate" />
+            <TextField source="role" />
+            <NumberField source="hoursPerWeek" />
+            <TextField source="costRate" />
+            <NumberField source="vacations" />
+            <TextField source="seniority" />
```

Comment:
please make it a dropdown. It can be fixed values for now. Same for role.
If you can, investigate how could the options be retrieved from an endpoint and see if that can be done.

Also check how to apply this to the other forms

Reply:
Completed by adding position and seniority as entities.